rust-runtime/smithy-eventstream/src/str_bytes.rs

```diff
@@ -106,34 +61,43 @@ impl From<&'static str> for StrBytes {
     fn from(value: &'static str) -> Self {
         StrBytes {
             bytes: Bytes::from(value),
-            valid_utf8: AtomicU8::new(State::Valid.as_u8()),
         }
     }
 }
 
-impl From<&'static [u8]> for StrBytes {
-    fn from(value: &'static [u8]) -> Self {
-        StrBytes {
-            bytes: Bytes::from(value),
-            valid_utf8: AtomicU8::new(State::Unknown.as_u8()),
+impl TryFrom<&'static [u8]> for StrBytes {
+    type Error = Utf8Error;
+
+    fn try_from(value: &'static [u8]) -> Result<Self, Self::Error> {
+        match std::str::from_utf8(value) {
+            Ok(_) => Ok(StrBytes {
+                bytes: Bytes::from(value),
+            }),
+            Err(err) => Err(err),
         }
     }
 }
 
-impl From<Vec<u8>> for StrBytes {
-    fn from(value: Vec<u8>) -> Self {
-        StrBytes {
-            bytes: Bytes::from(value),
-            valid_utf8: AtomicU8::new(State::Unknown.as_u8()),
+impl TryFrom<Vec<u8>> for StrBytes {
+    type Error = Utf8Error;
+
+    fn try_from(value: Vec<u8>) -> Result<Self, Self::Error> {
+        match std::str::from_utf8(&value[..]) {
+            Ok(_) => Ok(StrBytes {
+                bytes: Bytes::from(value),
+            }),
+            Err(err) => Err(err),
         }
     }
 }
 
-impl From<Bytes> for StrBytes {
-    fn from(bytes: Bytes) -> Self {
-        StrBytes {
-            bytes,
-            valid_utf8: AtomicU8::new(State::Unknown.as_u8()),
+impl TryFrom<Bytes> for StrBytes {
+    type Error = Utf8Error;
+
+    fn try_from(bytes: Bytes) -> Result<Self, Self::Error> {
```

Comment:
can you delegate these to all go through the same place? Either `TryFrom<Bytes>` or a `new` function?

That way we can audit that there is only one location that actually instantiates `StrBytes { bytes }` with the private constructor